Move database migration docs

trmartin4 opened the pull request:
## 📔 Objective

As written, our SQL Code Style [docs](https://contributing.bitwarden.com/contributing/code-style/sql) include mostly instructions on how to write migration scripts.  This PR moves those best practices closer to where they're applicable - the database migration docs - and also formats the documentation with expanders for easier consumption.

Note that I left the SQL Code Style docs as-is.  These will be expanded upon in https://github.com/bitwarden/contributing-docs/pull/648, and I didn't want to cause merge conflicts there.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation
  team

## 🦮 Reviewer guidelines

<!-- Suggested interactions but feel free to use (or not) as you desire! -->

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

withinfocus commented:
My other organization-related PR has merged so should we take a fresh look here?

trmartin4 commented:
@withinfocus I've updated these docs to incorporate your changes.

I also reached out to @mkincaid-bw because I think that his changes to the SQL Code Style doc will complement my changes very nicely; I've extracted all the migration-specific instructions into a separate doc, leaving the door open for more in-depth code style specific to SQL, which he's doing in https://github.com/bitwarden/contributing-docs/pull/648.

Would it be preferable for me to undo the changes to the Code Style doc in my PR?  I can do that before I open it for review, as what I put in there (https://restructure-sql-docs.contributing-docs.pages.dev/contributing/code-style/sql) is _very_ basic and based on some recommendations given for one of our database changes.

withinfocus commented:
I feel like these changes are SQL coding standards, not migration-writing ones. We should think of column additions, view recreations, etc. as just those standards and not focus on a specific usage. It would be ideal to me to have this in one document.
